includes/classes/order.php

```diff
@@ -1201,34 +1234,42 @@ function send_order_email($zf_insert_id = null)
         if ($this->content_type != 'virtual' && !$storepickup) {
             $email_order .= "\n" . EMAIL_TEXT_DELIVERY_ADDRESS . "\n" .
                 EMAIL_SEPARATOR . "\n" .
-                zen_address_label($_SESSION['customer_id'], $_SESSION['sendto'], false, '', "\n") . "\n";
+                zen_address_format($this->delivery['format_id'], $this->delivery, false, '', "\n") . "\n";
         }
         $email_order .= EMAIL_TEXT_TELEPHONE . $this->customer['telephone'] . "\n\n";
 
         //addresses area: Billing
         $email_order .= "\n" . EMAIL_TEXT_BILLING_ADDRESS . "\n" .
             EMAIL_SEPARATOR . "\n" .
-            zen_address_label($_SESSION['customer_id'], $_SESSION['billto'], false, '', "\n") . "\n\n";
+//            zen_address_label($_SESSION['customer_id'], $_SESSION['billto'], false, '', "\n") . "\n\n";
+            zen_address_format($this->billing['format_id'], $this->billing, false, '', "\n") . "\n\n";
         $html_msg['ADDRESS_BILLING_TITLE'] = EMAIL_TEXT_BILLING_ADDRESS;
-        $html_msg['ADDRESS_BILLING_DETAIL'] = zen_address_label($_SESSION['customer_id'], $_SESSION['billto'], true, '', "<br>");
+        $html_msg['ADDRESS_BILLING_DETAIL'] = zen_address_format($this->billing['format_id'], $this->billing, true, '', "<br>");
 
-        if (is_object($GLOBALS[$_SESSION['payment']])) {
+        $payment = '';
+        if (file_exists(DIR_WS_MODULES . 'payment/' . ($this->info['payment_module_code'] ?? 'NO_PAYMENT') . '.php')) {
+            require_once(DIR_WS_MODULES . 'payment/' . $this->info['payment_module_code'] . '.php');
+            require_once(DIR_WS_LANGUAGES . $_SESSION['language'] . '/modules/payment/' . $this->info['payment_module_code'] . '.php');
+            $payment = new $this->info['payment_module_code'];
+        }
+
+        if (!empty($payment)) {
             $cc_num_display = (isset($this->info['cc_number']) && $this->info['cc_number'] != '') ? /*substr($this->info['cc_number'], 0, 4) . */
                 str_repeat('X', (strlen($this->info['cc_number']) - 8)) . substr($this->info['cc_number'], -4) . "\n\n" : '';
             $email_order .= EMAIL_TEXT_PAYMENT_METHOD . "\n" .
                 EMAIL_SEPARATOR . "\n";
-            $payment_class = $_SESSION['payment'];
-            $email_order .= $GLOBALS[$payment_class]->title . "\n\n";
+            $email_order .= $payment->title . "\n\n";
             $email_order .= (isset($this->info['cc_type']) && $this->info['cc_type'] != '') ? $this->info['cc_type'] . ' ' . $cc_num_display . "\n\n" : '';
-            $email_order .= (isset($GLOBALS[$payment_class]->email_footer) && $GLOBALS[$payment_class]->email_footer) ? $GLOBALS[$payment_class]->email_footer . "\n\n" : '';
+            $email_order .= (isset($payment->email_footer) && $payment->email_footer) ? $payment->email_footer . "\n\n" : '';
```

Comment:
When this `$payment = new $this->info['payment_module_code']` is run, it creates a fresh new instance of the payment module, not the one that is currently active in the session.
Thus, it is likely that if the payment module has set a customized value (during current session processing) for the `email_footer` then that info will be lost here.

Reply:
i think my new commit will handle this concern.